src/main/java/org/cloudfoundry/promregator/config/OAuth2XSUAAAuthenticationConfiguration.java

```diff
@@ -1,45 +1,96 @@
 package org.cloudfoundry.promregator.config;
 
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.http.client.utils.URIBuilder;
+
 public class OAuth2XSUAAAuthenticationConfiguration {
-	private String tokenServiceURL;
-	
+	private String url;
+
+	private String cert_url;
+
 	private String client_id;
-	
+
 	private String client_secret;
 
-	private String scopes;
+	private String client_certificates;
+
+	private String client_key;
 
-	public String getTokenServiceURL() {
-		return tokenServiceURL;
+	private final Set<String> scopes = new HashSet<>();
+
+	public String getUrl() {
+		return url;
+	}
+
+	public void setTokenServiceURL(String url) {
+		try {
+			URI u = new URI(url);
+			setUrl(new URIBuilder().setScheme(u.getScheme()).setHost(u.getHost()).setPort(u.getPort())
+					.setPath(u.getPath().replaceAll("/oauth/token$", "")).build().toASCIIString());
```

Comment:
The approach is not able to deal with other suffixes in the path than `/oauth/token`. That is actually bad. But there is no general solution for that. Would be possible to drop the full path, but with that we would not be able to deal with prefixes in the path (e.g. `https://example.org/v1/oauth/token`).

In such a case it would be possible to provide `url` instead of `tokenServiceURL`.

Reply:
see also my comments above.
I don't worry the `/oauth/token` suffix too much. What worries me more is that the URL could also have some more specific URL attributes provided (e.g. due to this `/alias/xyz.eu10`). 
That is also why I came to the point that we should "sincerely try", but if we can't handle it, properly fail with a proper description for resolution.

Reply:
See my comments above.